integration/utmp_integration_test.go

```diff
@@ -115,6 +115,18 @@ func TestRootUTMPEntryExists(t *testing.T) {
 	t.Errorf("did not detect utmp entry within 5 minutes")
 }
 
+// TestUsernameLimit tests that the maximum length of usernames is a hard error.
+func TestUsernameLimit(t *testing.T) {
+	dir := t.TempDir()
+
+	// A 33 character long username.
+	username := "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	host := [4]int32{0, 0, 0, 0}
+	tty := os.NewFile(uintptr(5), "/proc/self/fd/0")
+	err := uacc.Open(dir+"/utmp", dir+"/wtmp", username, "localhost", host, tty)
+	require.Error(t, err)
+}
```

Comment:
Can we check that a 32 char username works and this isn't just throwing an error for other unknown reasons?

Reply:
added a test for that!